script/DeployL1OFTAdapter.s.sol

```diff
@@ -0,0 +1,92 @@
+/* solhint-disable no-console, gas-custom-errors */
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import {BaseScript} from "./BaseScript.s.sol";
+
+import {L1YnOFTAdapterUpgradeable} from "@/L1YnOFTAdapterUpgradeable.sol";
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {TransparentUpgradeableProxy} from
+    "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+
+import {console} from "forge-std/console.sol";
+
+// forge script script/DeployL1OFTAdapter.s.sol:DeployL1OFTAdapter \
+// --rpc-url ${rpc} --sig "run(string calldata)" ${path} \
+// --account ${deployerAccountName} --sender ${deployer} \
+// --broadcast --etherscan-api-key ${api} --verify
+
+contract DeployL1OFTAdapter is BaseScript {
+    L1YnOFTAdapterUpgradeable public l1OFTAdapter;
+
+    function run(string calldata _jsonPath) public {
+        _loadInput(_jsonPath);
+
+        require(currentDeployment.isL1 == true, "Must be L1 deployment");
+
+        bytes32 proxySalt = createL1YnOFTAdapterUpgradeableProxySalt(msg.sender);
+        bytes32 implementationSalt = createL1YnOFTAdapterUpgradeableSalt(msg.sender);
+
+        address deployer = msg.sender;
+
+        if (!isContract(currentDeployment.oftAdapter)) {
+            vm.broadcast();
+            address l1OFTAdapterImpl = address(
+                new L1YnOFTAdapterUpgradeable{salt: implementationSalt}(
+                    baseInput.l1ERC20Address, getData(block.chainid).LZ_ENDPOINT
+                )
+            );
+
+            bytes memory initializeData =
+                abi.encodeWithSelector(L1YnOFTAdapterUpgradeable.initialize.selector, deployer);
+
+            vm.broadcast();
+            l1OFTAdapter = L1YnOFTAdapterUpgradeable(
+                address(
+                    new TransparentUpgradeableProxy{salt: proxySalt}(l1OFTAdapterImpl, msg.sender, initializeData)
+                )
+            );
+
+            vm.broadcast();
+
+            address newOwner = getData(block.chainid).PROXY_ADMIN;
+            console.log("Changing owner for L1OFTAdapter to: %s", newOwner);
+            Ownable(getTransparentUpgradeableProxyAdminAddress(address(l1OFTAdapter))).transferOwnership(newOwner);
```

Comment:
we need to deploy a time lock here for the proxy upgradeability please on both sides

The timelock should be fully owned and controlled by the security council what is now called PROXY_ADMIN in your script.

Only apply this foor the upgrades